server/src/main/java/io/crate/planner/optimizer/symbol/rule/MoveReferenceCastToLiteralCastOnAnyOperatorsWhenRightIsReference.java

```diff
@@ -52,7 +53,7 @@ public MoveReferenceCastToLiteralCastOnAnyOperatorsWhenRightIsReference() {
             .with(f -> AnyOperator.OPERATOR_NAMES.contains(f.name()))
             .with(f -> f.arguments().get(0).symbolType().isValueOrParameterSymbol())
             .with(f -> f.arguments().get(1), typeOf(Function.class).capturedAs(castCapture)
-                .with(f -> f.isCast())
+                .with(f -> f.isCast() && CastMode.IMPLICIT.equals(f.castMode()))
```

Comment:
We need to limit this optimizer rule to implicit casts to fix the following scenario:
```
cr> select * from t;
+-------+-----+
| a     |   b |
+-------+-----+
| [1.1] | 1.1 |
+-------+-----+

cr> select * from t where 1 = any(a::int[]);
+---+---+
| a | b |
+---+---+
+---+---+
=> should not be empty

cr> select * from t where 1 = any(a::int[]);
+-------+-----+
| a     |   b |
+-------+-----+
| [1.1] | 1.1 | <-- expected result
+-------+-----+
```